sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -482,10 +482,10 @@ case class Cast(
 
   override def nullable: Boolean = child.nullable || Cast.forceNullable(child.dataType, dataType)
 
-  override def initQueryContext(): Option[SQLQueryContext] = if (ansiEnabled) {
-    Some(origin.context)
+  override def initQueryContext(): Array[SQLQueryContext] = if (ansiEnabled) {
```

Comment:
This API is for expression to provide its context. I don't think an expression can provide more than one context. I think keeping `Option[SQLQueryContext]` is better.

Reply:
> I think keeping Option[SQLQueryContext] is better.

@cloud-fan I reverted this back, and pass SQLQueryContext (can be null) to QueryExecutionErrors.